### DISCUSSION TITLE
Installation of DAGMC dependencies fails

### DISCUSSION THREAD

**Discussion by williamjsmith15** [svalinn/DAGMC#828] [Q&A] [answered]
Hi, when trying to install DAGMC to use with OpenMC, the MOAB install fails with error: 
config.status: error: cannot find input file: `Makefile.in' 
when running the command:
../src/configure --enable-optimize \
                   --enable-shared \
                   --disable-debug \
                   --with-hdf5=~/dagmc_bld/HDF5 \
                   --prefix=~/dagmc_bld/MOAB
I cannot find any fixes that already mention this and I have tried installing both from source and package manager 
following the guide here: https://svalinn.github.io/DAGMC/install/dependencies.html but with no luck either way. The install up to this point went fine with no issues.

Any help would be appreciated
Many Thanks
Will

**Comment by makeclean** [comment]
Hi @williamjsmith15  did you run `autoreconf -fi` before running configure?

**Reply by williamjsmith15** [reply]
Hi @makeclean, thanks for the quick reply!

I thought I had... but now running `autoconf -fi` in the dagmc_bld/MOAB/moab folder gives the error:
`autoheader: warning: cannot read config/MOABConfig.h.in: No such file or directory
configure.ac:78: installing 'config/compile'
configure.ac:14: installing 'config/missing'
configure.ac:1562: error: required file 'config/MOABConfig.h.in' not found
itaps/fbigeom/Makefile.am: installing 'config/depcomp'
autoreconf: error: automake failed with exit status: 1`

**Reply by makeclean** [reply]
Looks like autotools has had some failure, have you installed automake, autoconfig and libtool?

**Reply by williamjsmith15** [reply]
I already have automake, autoconf and libtool (and both fully updated) however cant find the package autoconfig - is this different to autoconf?

**Reply by pshriwise** [reply]
I've been running into this same problem. I think that the `MOABConfig.cmake.h.in` file's name was changed and this didn't get updated in the `configure.ac` file at some point. The following change in `configure.ac` fixed this problem for me. @williamjsmith15 if you try this out and it works for you as well I'll submit a PR to MOAB with a fix.

```diff --git a/configure.ac b/configure.ac
index 778b22670..81aeec84c 100644
--- a/configure.ac
+++ b/configure.ac
@@ -3,7 +3,7 @@
 ################################################################################
 AC_INIT([MOAB],[5.4.0],[moab-dev@mcs.anl.gov],[moab],[http://sigma.mcs.anl.gov])
 AC_CONFIG_SRCDIR([moab.make.in])
-AC_CONFIG_SRCDIR([config/MOABConfig.cmake.in])
+AC_CONFIG_SRCDIR([config/MOABConfig.cmake.h.in])
 AC_CONFIG_SRCDIR([examples/makefile.config.in])
 AC_CONFIG_MACRO_DIR([config])
 AC_CONFIG_AUX_DIR([config])
```

**Reply by williamjsmith15** [reply]
For some reason it still says it can't find this file, I have checked and the file is in the right location and named as you indicated, however it still exits with the same error.
```
configure.ac:1251: warning: The macro `AC_TRY_COMPILE' is obsolete.
configure.ac:1251: You should run autoupdate.
./lib/autoconf/general.m4:2847: AC_TRY_COMPILE is expanded from...
config/mesquite.m4:2: CONFIGURE_MESQUITE is expanded from...
configure.ac:1251: the top level
autoheader: warning: cannot read config/MOABConfig.h.in: No such file or directory
configure.ac:78: installing 'config/compile'
configure.ac:14: installing 'config/missing'
configure.ac:1570: error: required file 'config/MOABConfig.h.in' not found
itaps/fbigeom/Makefile.am: installing 'config/depcomp'
autoreconf: error: automake failed with exit status: 1
```

**Reply by williamjsmith15** [reply]
Only thing that I can see is different is I'm running version 5.1.0 and you are using 5.4.0...

**Reply by pshriwise** [reply]
Ah, yeah I was on the wrong track here. A MOAB developer indicated that this is a bug in older versions of MOAB. It seems that file was left out of the repo accidentally...

I've tested 5.3.0 and 5.4.0 (as you saw) and they no longer contain this issue. Would updating your MOAB version be an option? It's probably the easiest fix...

**Reply by williamjsmith15** [reply]
I can try updating the version, I am using it with Cubit and DAGMC to prepare CAD geometries for OpenMC simulations, I am assuming you don't know any issues with compatibility here, the only reason I was running that version being that the installation instructions on the DAGMC website said to do so.

**Reply by williamjsmith15** [reply]
That seems to have worked now, I will let you know if I run into any more issues

**Reply by pshriwise** [reply]
Great! Once you report back that you can build DAGMC successfully with this update, I'll go ahead and close this issue.

**Reply by williamjsmith15** [reply]
All good, DAGMC has built successfully, thanks for the assistance

**Comment by pshriwise** [comment, marked as answer]
Solved by updating MOAB version.
